beetsplug/rewrite.py

```diff
@@ -34,9 +34,7 @@ def rewriter(field, rules):
     def fieldfunc(item):
         value = item._values_fixed[field]
         for pattern, replacement in rules:
-            if pattern.match(value.lower()):
-                # Rewrite activated.
-                return replacement
+            return re.sub(pattern, replacement, value)
```

Comment:
Dang it, I don't know what I was thinking… I guess I shouldn't commit changes in the middle of the night anymore :)

I'll try to come up with something. But you're right - tests would make a lot of sense here.